src/components/UrlUtilsService.js

```diff
@@ -48,13 +48,44 @@ goog.provide('ga_urlutils_service');
                   /.*kmz$/.test(url));
         };
 
-        this.proxifyUrl = function(url) {
+        // Test using a head request if the remote resource enables CORS
+        this.isCorsEnabled = function(url) {
+          var deferred = $q.defer();
+          $http.head(url, { timeout: 1500 }).then(function(response) {
+            // If the head request responded successfully it means that CORS
+            // is enabled
+            deferred.resolve(true);
+          }, function() {
+            deferred.resolve(false);
+          });
+          return deferred.promise;
+        };
+
+        this.proxifyUrlInstant = function(url) {
           if (this.needsProxy(url)) {
             return gaGlobalOptions.ogcproxyUrl + encodeURIComponent(url);
           }
           return url;
         };
 
+        this.proxifyUrl = function(url) {
```

Comment:
I think the content of this function is a duplication of `needsProxy`, you could move the `isCorsEnabled` stuff in the `needsProxy` function then, promisifying :) the `needsProxy` function you keep the `proxifyUrl` function more understandable, and you get rid of `proxifyUrlInstant`:

```
this.proxifyUrl = function(url) {
  var defer = $q.defer();
  this.needsProxy(url).then(function() {
      defer.resolve(gaGlobalOptions.ogcproxyUrl + encodeURIComponent(url));
  }, function() {
      defer.resolve(url);
  })
  return defer.promise;
};
```